lib/nds_extract.rb

```diff
@@ -0,0 +1,90 @@
+# Provided, don't edit
+require 'directors_database'
+
+# A method we're giving you. This "flattens"  Arrays of Arrays so: [[1,2],
+# [3,4,5], [6]] => [1,2,3,4,5,6].
+
+def flatten_a_o_a(aoa)
+  result = []
+  i = 0
+
+  while i < aoa.length do
+    k = 0
+    while k < aoa[i].length do
+      result << aoa[i][k]
+      k += 1
+    end
+    i += 1
+  end
+
+  result
+end
+
+def movie_with_director_name(director_name, movie_data)
+  { 
+    :title => movie_data[:title],
+    :worldwide_gross => movie_data[:worldwide_gross],
+    :release_year => movie_data[:release_year],
+    :studio => movie_data[:studio],
+    :director_name => director_name
+  }
+end
+
+
+# Your code after this point
+
+def movies_with_director_key(name, movies_collection)
+  # GOAL: For each Hash in an Array (movies_collection), provide a collection
+  # of movies and a directors name to the movie_with_director_name method
+  # and accumulate the returned Array of movies into a new Array that's
+  # returned by this method.
```

Comment:
this goal is tough to interpret. Maybe providing an example input/output here or under RETURN would help

The solution to this seems to make sense once viewed, but to figured out how to get there is going to be tough for students